### PR TITLE
Allow to configure runtimes download options from goth_config

### PR DESCRIPTION
**Pull request by nieznanysprawiciel**
Features:
- Each additional artifact (runtimes, relay) can be configured individually from goth-config.yml
- Specify tag sub-string to fetch specific build from github repository
- Enable/disable using pre-releases
- Additional artifacts don't use pre-releases by default (we can test pre-released yagna with stable version of runtimes)

Example configuration file:
```
docker-compose:
  compose-file: "docker-compose.yml"
  docker-dir: "docker/"

  build-environment:
    artifacts:
      - name: "ya-runtime-vm"
        use-prerelease: true
        release-tag: "v0.3..*"
      - name: "ya-runtime-wasi"
        release-tag: "v0.2..*"
      - name: "ya-relay"
        use-prerelease: false
```

### CODE REVIEW

**Review by shadeofblue — APPROVED** [review]

File: test/unit/configuration/test-assets/goth-config-artifacts-override.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        use-prerelease: true
+        release-tag: "v0.3..*"
+      - name: "ya-runtime-wasi"
+        release-tag: "v0.2..*"
+      - name: "ya-relay-server"
+        use-prerelease: false
+
+  compose-log-patterns:
+    ethereum: ".*Wallets supplied."
+    zksync: ".*Running on http://.*:3030/.*"
```
Comment (kamirr) [inline test/unit/configuration/test-assets/goth-config-artifacts-override.yml:24]: None of those is valid now -- there are 3 ethereum containers (`ethereum-{mainnet,goerli,polygon}`) and the zksync container has been removed

**Review by kamirr — CHANGES_REQUESTED** [review]

**Comment by pwalski** [comment]
I updated example config from description, there should be `ya-relay` instead of `ya-relay-server`:
```
docker-compose:
  build-environment:
    artifacts:
      - name: "ya-relay"
        use-prerelease: false
```

File: test/unit/configuration/test-assets/goth-config-artifacts-override.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  docker-dir: "docker/"
+
+  build-environment:
+    artifacts:
+      - name: "ya-runtime-vm"
+        use-prerelease: true
+        release-tag: "v0.3..*"
+      - name: "ya-runtime-wasi"
+        release-tag: "v0.2..*"
+      - name: "ya-relay-server"
```
Comment (pwalski) [inline test/unit/configuration/test-assets/goth-config-artifacts-override.yml:19]:
```suggestion
      - name: "ya-relay"
```

**Review by pwalski — CHANGES_REQUESTED** [review]
There is a `ya-relay-server` artifact in test overrides.

File: test/unit/configuration/test-assets/goth-config-artifacts-override.yml (hunk 1 above)
Comment (nieznanysprawiciel) [reply test/unit/configuration/test-assets/goth-config-artifacts-override.yml]: I removed entries completely to avoid confusion

**Review by pwalski — APPROVED** [review]
It works on my branch locally (still has some problems on server, but these are unrelated).

**Review by kamirr — APPROVED** [review]

**Pull request merged**